diff --git a/webpack.config.js b/webpack.config.js
index 77b0338..8d35149 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -1,4 +1,5 @@
 const path = require('path');
+const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
 
 module.exports = {
     entry: './src/index.js',
@@ -30,10 +31,22 @@ module.exports = {
                         }
                     }
                 ]
+            },
+            {
+                test: /.tsx?$/,
+                use: [
+                    {
+                        loader: 'ts-loader',
+                        options: {
+                            transpileOnly: true
+                        }
+                    }
+                ]
             }
         ]
     },
     resolve: {
-        extensions: ['.js']
-    }
+        extensions: ['.js', '.ts', '.tsx']
+    },
+    plugins: [new ForkTsCheckerWebpackPlugin()]
 };
